lib/cdo/pardot.rb

```diff
@@ -469,6 +479,13 @@ def self.post_request(url, params)
 
     response = Net::HTTP.post_form(uri, params)
 
+    # Post to chat channel when known Pardot error occurs, then raise exception
+    # to be caught by calling functions.
+    if PARDOT_ERROR_CODES.include?(response.code)
```

Comment:
I suggest raising an error for any HTTP Response Code that is >= 500.  General practice when invoking a web service is that a 4XX Response Code indicates the client request was invalid and needs to be modified/corrected before being retransmitted.  5XX errors indicate a transient problem within the web service and should be retried.  Here’s some guidance that [AWS gives](https://docs.aws.amazon.com/general/latest/gr/api-retries.html) to people implementing their own HTTP client to invoke their web services:

> you should retry original requests that receive server (5xx) or throttling errors. However, client errors (4xx) indicate that you need to revise the request to correct the problem before trying again.

Reply:
Adding to Suresh's comment, one option for your consideration could be a `submit_prospect_batch_with_retry` function that wrap around the current `submit_prospect_batch`, and will retry when 500 errors are raised. (Kind of the same as `post_with_auth_retry` function.)

Reply:
Thanks for the feedback.  I would personally prefer to start with a narrow set of retries based on transient failures we've experienced rather than go too wide ranging, though I don't feel super strongly about that.

It's a fair point about making a new wrapper function, but given that `submit_prospect_batch` already has a retry mechanism which I'm just extending, I wanted to change as little code as possible.